### ISSUE TITLE
How to train the orientation inpainting and stroke inpainting model?

### ISSUE THREAD

**Issue by Jessicall** [tzt101/MichiGAN#5]
Nice work! I want to train the orientation inpainting and stroke inpainting model by myself,what should i do?
Another question is that, I see there is a blend network in the project, i specify the parameter '--use_blender' and use style,content,rgb,background,confidence,rgb loss to train the network,but the losses are not normal which their values are Nan.Have you trained the blend network successfully and could you give me any suggestion? What is the difference between using the blend network or not?
Thanks in advance.

**Comment by tzt101** [comment]
1. The orientation inpainting and stroke inpainting model are trained with the reconstructing mode (with GAN loss). Specifically, we randomly erase part of the hair area and train the model to reconstruct it. As for stroke inpainting model, the training data is generated as follows: we first randomly select a point, and create a stroke start at this point along the original orientation. Then, we erase the extended area around this stroke to generate the crosspending hole.
2. We previously used blender network to replace the background module and divided the whole synthesis process into two parts: hair synthesis and background blending. We first trained the hair synthesis module and then trained the blender networtk by fixing the parameters of hair synthesis module. But we found that this didn't work very well, so we finally use the background module and training the whole model jointly.
